### DISCUSSION TITLE
mass bit depth

### DISCUSSION THREAD

**Discussion by BendersRepublic** [ImageMagick/ImageMagick#6020] [Help]
### ImageMagick version

7.10-60

### Operating system, version and so on

Windows 11

### Description

Hello there, currently trying to figure out how to mass edit files and make them bit depth of 32, the cmd line is very confusing for me and I couldn't figure out how to do this

**Comment by BendersRepublic** [comment]
tried "magic *.png -depth 32" but got a missingargument error

**Comment by snibgo** [comment]
What do you mean by "bit depth of 32"? Do you want each channel to be 32 bits? Integer or floating-point? Or do you want 4 channels per pixel, with 8 bits per channel, integer? That last would be with a format of "PNG32".

For changing all the PNG images in a directory, I suggest using "`magick mogrify`", like this:
```
magick mogrify -format PNG32 *.png
```

**Reply by BendersRepublic** [reply]
I'm so sorry, I realize how little I know about this now lmao, I believe I need each channel to be in 32 bits, so it can reflect here: 
![image](https://user-images.githubusercontent.com/108898499/216158912-34e41259-f1af-4523-b63d-f2960f41d5a7.png)
